dojo/tools/bugcrowd_api/api_client.py

```diff
@@ -0,0 +1,140 @@
+            params = params_default
+            params["filter[target]"] = target
+            params_encoded = urlencode(params)
+        else:
+            params_encoded = urlencode(params_default)
+
+        next = "{}/submissions?{}".format(self.bugcrowd_api_url, params_encoded)
+        while next != "":
+            response = self.session.get(url=next)
+
+            if response.ok:
+                data = response.json()
+                if len(data["data"]) != 0:
+                    yield data["data"]
+
+                # When we hit the end of the submissions, break out
+                if len(data["data"]) == 0:
+                    next = ""
+                    break
+
+                # Otherwise, keep updating next link
+                next = "{}{}".format(self.bugcrowd_api_url, data["links"]["next"])
+            else:
+                next = "over"
+                raise Exception(
+                    "Unable to get bugcrowd submissions due to {} - {}".format(
+                        response.status_code, response.content.decode("utf-8")
+                    )
+                )
+
+    def test_connection(self):
+        # Request programs
+        response_programs = self.session.get(
+            url="{}/programs".format(self.bugcrowd_api_url)
+        )
+
+        # Request submissions to validate the org token
+        response_subs = self.session.get(
+            url="{}/submissions".format(self.bugcrowd_api_url)
+        )
+
+        if response_programs.ok and response_subs.ok:
+            data = response_programs.json().get("data")
+            data_subs = response_subs.json().get("meta")
+            total_subs = str(data_subs["total_hits"])
+
+            progs = list(filter(lambda prog: prog["type"] == "program", data))
+            program_names = ", ".join(
+                list(map(lambda p: p["attributes"]["code"], progs))
+            )
+            # Request targets to validate the org token
+            response_targets = self.session.get(
+                url="{}/targets".format(self.bugcrowd_api_url)
+            )
+            if response_targets.ok:
+                data_targets = response_targets.json().get("data")
+                targets = list(
+                    filter(lambda prog: prog["type"] == "target", data_targets)
+                )
+                target_names = ", ".join(
+                    list(map(lambda p: p["attributes"]["name"], targets))
+                )
+                return f'With {total_subs} submissions, you have access to the "{ program_names }" programs, \
+                    you can use these as Service key 1 for filtering submissions \
+                        You also have targets "{ target_names }" that can be used in Service key 2'
+        else:
+            raise Exception(
+                "Connection failed (error: {} - {})".format(
+                    response_subs.status_code, response_subs.content.decode("utf-8")
+                )
+            )
```

Comment:
use `response.raise_for_status()` at line 87 and 92 instead.

Reply:
Done